DiabloII.Application/Services/Hals/Suggestions/SuggestionHalService.cs

```diff
@@ -0,0 +1,18 @@
+﻿using System.Net.Http;
+using DiabloII.Application.Extensions;
+using DiabloII.Application.Responses.Suggestions;
+using Halcyon.HAL;
+using Microsoft.AspNetCore.Mvc;
+
+namespace DiabloII.Application.Services.Hals.Suggestions
+{
+    public class SuggestionHalService : BaseHalService
+    {
+        public HALResponse AddLinks(SuggestionDto dto, ControllerBase controller)
+        {
+            var halResponse = ToHalResponse(controller, dto);
+
+            return halResponse.AddLink(controller, "suggestion_create", HttpMethod.Post);
```

Comment:
To generate the other sub links correlated to the UI visibilities, I'll need the information related to the user link the userId.